### PR TITLE
Update `docs/source-pytorch/common/lightning_module.rst`

### PR DESCRIPTION
**Pull request by JustinGoheen**
## What does this PR do?

Updates examples per #18327 

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--18451.org.readthedocs.build/en/18451/

### CODE REVIEW

File: docs/source-pytorch/common/lightning_module.rst [hunk 1]
```diff
@@ -84,39 +84,45 @@ Here are the only required methods.
 
 .. code-block:: python
 
-    import lightning.pytorch as pl
-    import torch.nn as nn
-    import torch.nn.functional as F
+    import pytorch_lightning as pl
```
Comment (Borda) [inline docs/source-pytorch/common/lightning_module.rst:87]: I think that the imports shall be consistent, so keep the import as it was

Comment (JustinGoheen) [reply docs/source-pytorch/common/lightning_module.rst]: updated in [648e829](https://github.com/Lightning-AI/lightning/pull/18451/commits/648e8295d62b3b6019388ed5119aa64be7d89143)

**Comment by JustinGoheen** [comment]
updated page on rtfd staging: [LightningModule](https://pytorch-lightning--18451.org.readthedocs.build/en/18451/common/lightning_module.html)

File: docs/source-pytorch/common/lightning_module.rst [hunk 2]
```diff
@@ -370,17 +378,23 @@ There are two ways to call ``test()``:
...
 
     # call after training
-    trainer = Trainer()
-    trainer.fit(model)
+    trainer = pl.Trainer()
+    trainer.fit(model=model, train_dataloaders=dataloader)
 
     # automatically auto-loads the best weights from the previous run
-    trainer.test(dataloaders=test_dataloader)
+    trainer.test(dataloaders=trainer.test_dataloaders)
```
[the remark this answers is not in the document]
Comment (JustinGoheen) [reply docs/source-pytorch/common/lightning_module.rst]: changed in [a6cbab6](https://github.com/Lightning-AI/lightning/pull/18451/commits/a6cbab69ad9abd00e373fb812cea5ddcb8cec57b)

File: docs/source-pytorch/common/lightning_module.rst [hunk 3]
```diff
@@ -85,38 +85,44 @@ Here are the only required methods.
...
+        def forward(self, batch):
+            input, target = batch
+            return self.model(input.view(1, -1), target.view(1, -1))
 
         def training_step(self, batch, batch_idx):
-            x, y = batch
-            y_hat = self(x)
-            loss = F.cross_entropy(y_hat, y)
+            input, target = batch
+            output = self.model(input, target)
```
[the remark this answers is not in the document]
Comment (JustinGoheen) [reply docs/source-pytorch/common/lightning_module.rst]: changed in [a6cbab6](https://github.com/Lightning-AI/lightning/pull/18451/commits/a6cbab69ad9abd00e373fb812cea5ddcb8cec57b)

File: docs/source-pytorch/common/lightning_module.rst [hunk 4]
```diff
@@ -85,38 +85,44 @@ Here are the only required methods.
...
+        def __init__(self, vocab_size):
             super().__init__()
-            self.l1 = nn.Linear(28 * 28, 10)
+            self.model = Transformer(vocab_size=vocab_size)
 
-        def forward(self, x):
-            return torch.relu(self.l1(x.view(x.size(0), -1)))
+        def forward(self, batch):
+            input, target = batch
+            return self.model(input.view(1, -1), target.view(1, -1))
```
[the remark this answers is not in the document]
Comment (JustinGoheen) [reply docs/source-pytorch/common/lightning_module.rst]: changed in [a6cbab6](https://github.com/Lightning-AI/lightning/pull/18451/commits/a6cbab69ad9abd00e373fb812cea5ddcb8cec57b)

File: docs/source-pytorch/common/lightning_module.rst [hunk 5]
```diff
@@ -85,38 +85,44 @@ Here are the only required methods.
...
-    model = LitModel()
+    from lightning.pytorch.demos import WikiText2
+    from torch.utils.data import DataLoader
+
+    dataset = WikiText2()
+    dataloader = DataLoader(dataset)
+    model = LightningTransformer(vocab_size=dataset.vocab_size)
 
-    trainer.fit(model, train_dataloaders=train_loader)
+    trainer = pl.Trainer(fast_dev_run=True)
```
[the remark this answers is not in the document]
Comment (JustinGoheen) [reply docs/source-pytorch/common/lightning_module.rst]: `pl.Trainer(fast_dev_run=100)` used in [a6cbab6](https://github.com/Lightning-AI/lightning/pull/18451/commits/a6cbab69ad9abd00e373fb812cea5ddcb8cec57b)

**Review by awaelchli — APPROVED** [review]

File: docs/source-pytorch/common/lightning_module.rst [hunk 6]
```diff
@@ -85,38 +85,43 @@ Here are the only required methods.
...
-        def forward(self, x):
-            return torch.relu(self.l1(x.view(x.size(0), -1)))
+        def forward(self, input, target):
+            return self.model(input, target)
 
         def training_step(self, batch, batch_idx):
-            x, y = batch
-            y_hat = self(x)
-            loss = F.cross_entropy(y_hat, y)
+            input, target = batch
```
Comment (Borda) [inline docs/source-pytorch/common/lightning_module.rst:102]: lets not use python keywords like `input`

File: docs/source-pytorch/common/lightning_module.rst [hunk 7]
```diff
@@ -85,38 +85,43 @@ Here are the only required methods.
...
 
 .. code-block:: python
 
-    train_loader = DataLoader(MNIST(os.getcwd(), download=True, transform=transforms.ToTensor()))
-    trainer = pl.Trainer(max_epochs=1)
-    model = LitModel()
+    from lightning.pytorch.demos import WikiText2
+    from torch.utils.data import DataLoader
+
+    dataset = WikiText2()
```
Comment (Borda) [inline docs/source-pytorch/common/lightning_module.rst:117]:
```suggestion
    dataset = pl.demos.WikiText2()
```

File: docs/source-pytorch/common/lightning_module.rst [hunk 8]
```diff
@@ -85,38 +85,43 @@ Here are the only required methods.
...
+            return torch.optim.SGD(self.model.parameters(), lr=0.1)
 
 Which you can train by doing:
 
 .. code-block:: python
 
-    train_loader = DataLoader(MNIST(os.getcwd(), download=True, transform=transforms.ToTensor()))
-    trainer = pl.Trainer(max_epochs=1)
-    model = LitModel()
+    from lightning.pytorch.demos import WikiText2
```
Comment (Borda) [inline docs/source-pytorch/common/lightning_module.rst:114]:
```suggestion
```

File: docs/source-pytorch/common/lightning_module.rst [hunk 9]
```diff
@@ -370,17 +377,23 @@ There are two ways to call ``test()``:
...
-    trainer = Trainer()
+    model = LightningTransformer.load_from_checkpoint(PATH)
+    dataset = WikiText2()
+    test_dataloader = DataLoader(dataset)
+    trainer = pl.Trainer()
     trainer.test(model, dataloaders=test_dataloader)
 
+.. note::
+    `WikiText2` is used in a manner that does not create a train, test, val split. This is done for illustrative purposes only.
+    A proper split can be created in :meth:`lightning.pytorch.core.module.LightningModule.setup` or :meth:`lightning.pytorch.core.module.LightningDataModule.setup`.
```
Comment (Borda) [inline docs/source-pytorch/common/lightning_module.rst:394]:
```suggestion
    A proper split can be created in :meth:`lightning.pytorch.core.LightningModule.setup` or :meth:`lightning.pytorch.core.LightningDataModule.setup`.
```

File: docs/source-pytorch/common/lightning_module.rst [hunk 10]
```diff
@@ -440,15 +447,17 @@ There are two ways to call ``predict()``:
...
+    trainer.fit(model=model, train_dataloaders=dataloader)
 
     # automatically auto-loads the best weights from the previous run
     predictions = trainer.predict(dataloaders=predict_dataloader)
 
     # or call with pretrained model
-    model = MyLightningModule.load_from_checkpoint(PATH)
-    trainer = Trainer()
+    model = LightningTransformer.load_from_checkpoint(PATH)
+    dataset = WikiText2()
```
Comment (Borda) [inline docs/source-pytorch/common/lightning_module.rst:457]:
```suggestion
    dataset = pl.demos.WikiText2()
```

**Review by Borda — APPROVED** [review]

File: docs/source-pytorch/common/lightning_module.rst (hunk 6 above)
Comment (JustinGoheen) [reply docs/source-pytorch/common/lightning_module.rst]: updated in [93a94fd](https://github.com/Lightning-AI/lightning/pull/18451/commits/93a94fd330b8da663797b727dc09293d37be46d5)

File: docs/source-pytorch/common/lightning_module.rst [hunk 11]
```diff
@@ -440,15 +446,17 @@ There are two ways to call ``predict()``:
...
+    trainer.fit(model=model, train_dataloaders=dataloader)
 
     # automatically auto-loads the best weights from the previous run
     predictions = trainer.predict(dataloaders=predict_dataloader)
 
     # or call with pretrained model
-    model = MyLightningModule.load_from_checkpoint(PATH)
-    trainer = Trainer()
+    model = LightningTransformer.load_from_checkpoint(PATH)
+    dataset = pl.demos.WikiText2()
```
Comment (awaelchli) [inline docs/source-pytorch/common/lightning_module.rst:457]: @Borda This doesn't work.
@JustinGoheen Could you please revert it in #18452?

File: docs/source-pytorch/common/lightning_module.rst [hunk 12]
```diff
@@ -85,38 +85,42 @@ Here are the only required methods.
...
 Which you can train by doing:
 
 .. code-block:: python
 
-    train_loader = DataLoader(MNIST(os.getcwd(), download=True, transform=transforms.ToTensor()))
-    trainer = pl.Trainer(max_epochs=1)
-    model = LitModel()
+    from torch.utils.data import DataLoader
+
+    dataset = pl.demos.WikiText2()
```
Comment (awaelchli) [inline docs/source-pytorch/common/lightning_module.rst:116]: Here too

File: docs/source-pytorch/common/lightning_module.rst (hunk 11 above)
Comment (JustinGoheen) [reply docs/source-pytorch/common/lightning_module.rst:457]: this is changed in [6671aa6](https://github.com/Lightning-AI/lightning/pull/18452/commits/6671aa6215041d1f8598854e39fd7a5c5330cb69)

File: docs/source-pytorch/common/lightning_module.rst (hunk 12 above)
Comment (JustinGoheen) [reply docs/source-pytorch/common/lightning_module.rst:116]: this is changed in [6671aa6](https://github.com/Lightning-AI/lightning/pull/18452/commits/6671aa6215041d1f8598854e39fd7a5c5330cb69)
